src/bin/pgcopydb/copydb.c

```diff
@@ -679,23 +679,11 @@ copydb_init_specs(CopyDataSpec *specs,
 
 		.sourceTableHashByOid = NULL
 	};
-
-	/* initialize the connection strings */
-	if (!IS_EMPTY_STRING_BUFFER(options->source_pguri))
+	if(tmpCopySpecs.source_pguri == NULL || tmpCopySpecs.target_pguri == NULL || tmpCopySpecs.sourceSnapshot.pguri == NULL)
 	{
-		strlcpy(tmpCopySpecs.source_pguri,
-				options->source_pguri,
-				MAXCONNINFO);
-
-		strlcpy(tmpCopySpecs.sourceSnapshot.pguri,
-				options->source_pguri,
-				MAXCONNINFO);
+		return false;
```

Comment:
As we never test the result of `strdup()`, which is NULL when there is not enough memory, then having a test here seems okay. That said, we should `log_error(ALLOCATION_FAILED_ERROR)` before returning false.

Nitpicking: please install the code formatting tools we use, same as Citus, and cut your lines down to 80 chars max. It's old, but the physical reality of concern is that eyes tire more in horizontal movements than in vertical movements. Well this is not a good place for line length arguments, let's just all apply the same limit: 80.